### PR TITLE
Fix fatal error on loading extension page when an extension has been deleted

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Softens the scenario where an installed extension has been deleted, allowing the extensions page to still load

Before
----------------------------------------
1) install an extension
2) delete the installed extension
3) load the extension page
Screen Shot 2020-03-12 at 12 22 08 PM

After
----------------------------------------
Screen Shot 2020-03-12 at 12 35 37 PM

Technical Details
----------------------------------------
Since we added the concept of 'hidden' tags we are loading them in a way where the exceptions are not caught. This means
that the whole page cannot load - which is needed to disable the missing extension

As the hidden tag / core extension concept was added in 5.24 I'm targetting 5.24

Comments
----------------------------------------
The messages are possibly a bit overkill but also not necessarily in a bad way

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this  please

**Comment by seamuslee001** [comment]
So in theory lets say you deleted a payment processor extension would you not want that to be a hard fail but maybe that hard fails in other ways as well. Not opposed to this but would appreciate @totten comment on this

File: CRM/Extension/Mapper.php [hunk 1]
```diff
@@ -441,7 +442,16 @@ public function getAllTags() {
    */
   public function getAllInfos() {
     foreach ($this->container->getKeys() as $key) {
-      $this->keyToInfo($key);
+      try {
+        $this->keyToInfo($key);
+      }
+      catch (CRM_Extension_Exception_ParseException $e) {
+        CRM_Core_Session::setStatus(ts('Parse error in extension: %1', [
```
Comment (mattwire) [inline CRM/Extension/Mapper.php]: This error sounds a bit developer-y?

Comment (eileenmcnaughton) [reply CRM/Extension/Mapper.php]: Yeah - but it was already in use elsewhere - I just copied it up a level

**Comment by seamuslee001** [comment]
I guess my thinking was more along the lines of is the fact that we would be allowing the system work smoother are there any consequences of that practice. I don't think so but just thought worth verbalising that thinking. I agree with matt that the message seems to be  bit developery  but it is already there so at least we are being consistent. I'm going to merge this as the code looks correct to me and the unit tests cover a number of interactions with exts

**Comment by eileenmcnaughton** [comment]
thanks @seamuslee001

**Pull request merged**